Find a functionally equivalent code:android.util.LongSparseLongArray.append:COMMENT
Method Modifier: public      
Comment:/**
 * Puts a key/value pair into the array, optimizing for the case where
 * the key is greater than all existing keys in the array.
 */

@@ -3,13 +3,9 @@
         put(key, value);
         return;
     }
-    int pos = mSize;
-    if (pos >= mKeys.length) {
-        growKeyAndValueArrays(pos + 1);
-    }
-    mKeys[pos] = key;
-    mValues[pos] = value;
-    mSize = pos + 1;
+    mKeys = GrowingArrayUtils.append(mKeys, mSize, key);
+    mValues = GrowingArrayUtils.append(mValues, mSize, value);
+    mSize++;
 }

